prescient.el

```diff
@@ -982,6 +982,307 @@ actually just the input, in which case nothing happens."
     "Filtering using prescient.el.
 For sorting, see the function `prescient-completion-sort'."))
 
+;;;;; Minor mode for the completion style.
+
+;; This minor mode makes it easier to have a Vertico or Corfu mode
+;; just focus on those interfaces.
+
+(defconst prescient--completion-settings-vars
+  '( completion-styles completion-category-overrides
+     completion-category-defaults)
+  "Variables that are changed when using `prescient-completion-mode'.")
+
+(defvar prescient--completion-recommended-styles '(prescient basic)
+  "Recommended completions style for using prescient.")
+
+(defvar prescient--completion-old-styles nil
+  "Previous value of `completion-styles'.")
+
+(defvar prescient--completion-recommended-overrides
+  '(;; Include `partial-completion' to enable wildcards and
+    ;; partial paths.
+    (file (styles basic partial-completion))
+    ;; Eglot forces `flex' by default.
+    (eglot (styles prescient basic)))
+  "Recommended completion-category overrides for using prescient.")
+
+(defvar prescient--completion-old-overrides nil
+  "Previous value of `completion-category-overrides'.")
+
+(defvar prescient--completion-old-defaults nil
+  "Previous value of `completion-category-defaults'.")
+
+(defconst prescient--completion-old-vars
+  '( prescient--completion-old-styles
+     prescient--completion-old-overrides
+     prescient--completion-old-defaults)
+  "Variables used to store old settings.")
+
+;; We keep these "settings" functions separate so that we can use them
+;; in the integration packages without running the entire mode
+;; function.
+
+(defun prescient--completion-apply-completion-settings ()
+  "Apply the recommended settings."
+  (setq completion-styles prescient--completion-recommended-styles
+        completion-category-defaults nil)
+
+  (cl-symbol-macrolet
```

Comment:
Personally I wouldn't use `cl-symbol-macrolet` since it isn't widely used and leads to non-obvious code. Of course this is only a matter of style. I've not found other uses of `cl-symbol-macrolet` in the code base, so maybe @raxod502 usually avoids this macro?

Reply:
I'd never heard of it personally, but if you think it's a good function I certainly won't object. No feelings one way or the other.